src/core/thread/network_diagnostic.cpp

```diff
@@ -77,20 +81,54 @@ void NetworkDiagnostic::SetReceiveDiagnosticGetCallback(otReceiveDiagnosticGetCa
     mReceiveDiagnosticGetCallbackContext = aCallbackContext;
 }
 
+ThreadError NetworkDiagnostic::SendEmptyAck(Coap::Header &aHeader, const Ip6::MessageInfo &aMessageInfo)
```

Comment:
Yes, i agree.

Would it be better idea first to accept this PR and after that create right away a new one, since this change will affect not only DIAG module but also other modules/messages like: Address Release, Address Notification, Server Data Notification and many others in MeshCop.

Also since this is Thread specific pattern Ntf/Qry/Ans_CON i think that current openthread-coap.h API should not be extended for this extra function. Thoughts?